src/schema/users.ts

```diff
@@ -1649,7 +1650,17 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       __,
       ctx: AuthContext,
       info,
-    ): Promise<GQLUserStreak> => {
+    ): Promise<GQLUserStreak | null> => {
+      // Check if user has opted out of reading streaks
+      const settings = await ctx.con.getRepository(Settings).findOne({
+        where: { userId: ctx.userId },
+        select: ['optOutReadingStreak'],
+      });
+
+      if (settings?.optOutReadingStreak) {
+        return null;
+      }
```

Comment:
Not the best solution since we are introducing another query that needs to run before querying streak

We need to do it in parallel, can you maybe add this as join in `getUserStreakQuery` and then select `settings.optOutReadingStreak` from there.

Then you can do `streak.optOutReadingStreak` to return null in places you want.